src/error.rs

```diff
@@ -197,6 +197,16 @@ impl fmt::Display for Error {
     }
 }
 
+impl Into<io::Error> for Error {
+    fn into(self) -> io::Error {
+        if self.is_timeout() {
+            io::Error::new(io::ErrorKind::TimedOut, self)
+        } else {
+            io::Error::new(io::ErrorKind::Other, self)
+        }
```

Comment:
Was this inspired by #1067? Thanks for having a look at it anyway :-).

I wonder if it would make sense to employ further checks, like:
* `is_connect` -> `NotConnected`
* `is_decode` -> `InvalidData`

Reply:
Just a coincidence from my own work :sweat_smile:. @seanmonstar was on the fence about the change when we chatted about it on discord so I just did the bare minimum to get some opinions. If there's an interest in getting it merged I can refine it with better checks and address the other comment